### PR TITLE
Write colored log output (at least on UNIX systems)

### PR DESCRIPTION
**Pull request by pointhi**
Currently, spotting the error line in bigger amounts of terminal output is quite a pain.

This commit adds the new function ```log_error```, which handles such message in an improved way:
* write message to stderr instead of stdout
* print the string in red color on the terminal (at least for unix terminals now)

### CODE REVIEW

File: mx.py [hunk 1]
```diff
@@ -9341,6 +9341,21 @@ def log(msg=None):
...
 
+def log_error(msg=None):
+    """
+    Write an error message to the console.
+    All script output goes through this method thus allowing a subclass
+    to redirect it.
+    """
+    if msg is None:
+        sys.stderr.write(os.linesep)
+    elif sys.platform.startswith('linux') or\
```
Comment (dougxc) [inline mx.py]: no need to split such short lines

File: mx.py [hunk 2]
```diff
@@ -9341,6 +9341,21 @@ def log(msg=None):
...
+         sys.platform in ['darwin', 'freebsd']:
+        # On unix systems, we can use ANSI escape sequences
+        sys.stderr.write('\033[91m' + msg.encode('utf8') + '\033[0m' + os.linesep)
```
Comment (dougxc) [inline mx.py]: How does this look in captured log files? Maybe we should only colorize if in an interactive console.

**Comment by pointhi** [comment]
@dougxc thanks for your input.

I changed the PR accordingly

**Comment by dougxc** [comment]
I assume the encoding being performed is to mimic what the `print` statement does. In that case, let's just use `print` directly:
```
    if msg is None:
        print >> sys.stderr
    elif isUnix and sys.stderr.isatty():
        # On unix systems, we can use ANSI escape sequences
        print >> sys.stderr, '\033[91m' + msg + '\033[0m'
    else:
        print >> sys.stderr, msg
```
If you're ok with that, I'll integrate it.

**Comment by pointhi** [comment]
No problem, you can integrate it.

**Pull request merged**